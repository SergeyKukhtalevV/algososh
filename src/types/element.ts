import {ElementStates} from "./element-states";

export type TElement = {
  value: string;
  color: ElementStates
}

export type TNumber = {
  value: number;
  color: ElementStates
}
