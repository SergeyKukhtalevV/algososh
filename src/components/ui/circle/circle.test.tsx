import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";
import renderer from "react-test-renderer";
import React from 'react';

describe('CircleComponent', () => {
  it('Circle without letter', ()=> {
    const circle = renderer
      .create(<Circle/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with letters', ()=> {
    const circle = renderer
      .create(<Circle letter={'abcd'}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with head', ()=> {
    const circle = renderer
      .create(<Circle head={'abcd'}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with React element in head', ()=> {
    const circle = renderer
      .create(<Circle head={<Circle />}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with tail', ()=> {
    const circle = renderer
      .create(<Circle tail={'abcd'}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with React element in tail', ()=> {
    const circle = renderer
      .create(<Circle tail={<Circle />}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with index', ()=> {
    const circle = renderer
      .create(<Circle index={1}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle with props isSmall ===  true', ()=> {
    const circle = renderer
      .create(<Circle isSmall={true}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle default', ()=> {
    const circle = renderer
      .create(<Circle/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle changing', ()=> {
    const circle = renderer
      .create(<Circle state={ElementStates.Changing}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Circle modified', ()=> {
    const circle = renderer
      .create(<Circle state={ElementStates.Modified}/>)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
});
