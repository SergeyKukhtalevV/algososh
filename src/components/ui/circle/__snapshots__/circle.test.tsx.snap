// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleComponent Circle changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle with React element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle with React element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`CircleComponent Circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    abcd
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`CircleComponent Circle with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abcd
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle with props isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CircleComponent Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    abcd
  </div>
</div>
`;

exports[`CircleComponent Circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
